AutoCollection/WebSnippet.ts

```diff
@@ -0,0 +1,146 @@
+import * as path from "path";
+import http = require("http");
+import fs = require('fs');
+
+import Logging = require("../Library/Logging");
+import TelemetryClient = require("../Library/TelemetryClient");
+
+class WebSnippet {
+
+    public static INSTANCE: WebSnippet;
+
+    private static _snippet: string;
+    private static _aiUrl: string;
+    private _isEnabled: boolean;
+    private _isInitialized: boolean;
+
+    constructor(client: TelemetryClient) {
+        if (!!WebSnippet.INSTANCE) {
+            throw new Error("Web snippet injection should be configured from the applicationInsights object");
+        }
+
+        WebSnippet.INSTANCE = this;
+        // AI URL used to validate if snippet already included
+        WebSnippet._aiUrl = " https://js.monitor.azure.com/scripts/b/ai.2";
+
+        let snippetPath = path.resolve(__dirname, "../../node_modules/applicationinsight-web-snippet/snippet/snippet.min.js");
+        if (client.config.isDebugWebSnippet) {
+            snippetPath = path.resolve(__dirname, "../../node_modules/applicationinsight-web-snippet/snippet/snippet.js");
+        }
+        fs.readFile(snippetPath, function (err, snippet) {
+            if (err) {
+                Logging.warn("Failed to load AI Web snippet. Ex:" + err);
+            }
+            WebSnippet._snippet = snippet.toString().replace("INSTRUMENTATION_KEY", client.config.instrumentationKey);
+        });
+    }
+
+    public enable(isEnabled: boolean) {
```

Comment:
When do users call this public method?  Seems like you want to support "disable" functionality as well!  If so, there isn't much disabling going on other than setting the flag.  Also, I don't see this._isEnabled used anywhere so if someone sets this to false, what exactly will happen?

Reply:
This will be used by applicationInsights.ts during init or reset process. it should be enable/disable like all other configs